cfy_manager/components/postgresql_server/postgresql_server.py

```diff
@@ -216,29 +233,174 @@ def _update_postgres_password(self):
         config[POSTGRESQL_SERVER][POSTGRES_PASSWORD] = '<removed>'
         logger.notice('postgres password successfully updated')
 
-    def _configure(self):
-        files.copy_notice(POSTGRESQL_SERVER)
-        self._init_postgresql_server()
-        enable_remote_connections = \
-            config[POSTGRESQL_SERVER][ENABLE_REMOTE_CONNECTIONS]
-        self._update_configuration(enable_remote_connections)
-        if config[POSTGRESQL_SERVER][POSTGRES_PASSWORD]:
-            self._update_postgres_password()
-
-        systemd.restart(SYSTEMD_SERVICE_NAME, append_prefix=False)
-        systemd.verify_alive(SYSTEMD_SERVICE_NAME, append_prefix=False)
+    def _etcd_is_running(self):
+        status = common.run(['pgrep', 'etcd'], ignore_failures=True)
```

Comment:
systemctl status?
(it has different retcodes depending on whether the service is down or not declared yet)

Reply:
It's not so much about checking whether it's been declared as whether it's actually started the process, which I think this might be better for that systemctl status as this has much clearer semantics- if it returns non-zero, it didn't find the process in question.

Reply:
the "declared" comment was because i was expecting you to mention that you dont want it to fail if the service isnt created yet :P it doesn't matter otherwise
i do think services are the right level of abstraction here, not processes. Especially since this is polled after a `systemctl start`

Reply:
is etcd not considered "started" before it actually connects?